### PR TITLE
enhancement: error handling + exp flag

### PR DESCRIPTION
**Pull request by AmeanAsad**
### Changes:
- Add experimental flag to load nodes.
- Ignores fallback on certain errors and status codes. 
- Test cases for those fallback edge cases

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -54,7 +56,7 @@ export class Saturn {
       this._monitorPerformanceBuffer()
     }
     this.storage = this.opts.storage || memoryStorage()
-    this.loadNodesPromise = this._loadNodes(this.opts)
+    this.loadNodesPromise = this.opts.experimental ? this._loadNodes(this.opts) : null
```
Comment (AmeanAsad) [inline src/client.js:59]: @DiegoRBaquero added this to only load nodes for the % we desire per a / b testing.

File: src/client.js [hunk 2]
```diff
@@ -81,7 +83,7 @@ export class Saturn {
       }
     }
 
-    const origins = options.origins
+    const origins = options.origins || [options.cdnURL]
```
Comment (guanzo) [inline src/client.js:86]: if `options.origins` is an empty array it's still truthy, is that ok here?

Comment (DiegoRBaquero) [inline src/client.js:86]:
```suggestion
    const origins = options.origins ?? [options.cdnURL]
```

**Review by DiegoRBaquero — APPROVED** [review]

File: src/client.js (hunk 2 above)
Comment (AmeanAsad) [reply src/client.js]: @DiegoRBaquero updated this with your suggestion and eric's comment above

Comment (AmeanAsad) [reply src/client.js]: not ok, thanks for flagging. I've added more logic to handle this better.

File: src/client.js [hunk 3]
```diff
@@ -81,7 +83,10 @@ export class Saturn {
       }
     }
 
-    const origins = options.origins
+    let origins = options.origins
+    if (!origins || origins.length === 0) {
+      origins = [options.url] ?? [options.cdnURL]
```
Comment (DiegoRBaquero) [inline src/client.js:88]: `[options.url]` won't ever be null/undefined so `[options.cdnURL]` is a dead path

Comment (AmeanAsad) [reply src/client.js]: updated again

**Review by DiegoRBaquero — APPROVED** [review]

**Review by guanzo — APPROVED** [review]
